Dead code detection job fails with 'Argument list too long' when issue count is high

## Problem

The TypeScript dead code detection job fails in CI when the number of detected issues exceeds a certain threshold, causing the script to hit the system's `ARG_MAX` limit.

### Error Details

**Job**: https://github.com/kaeawc/auto-mobile/actions/runs/20904545178/job/60055733951

```
🔍 Running TypeScript dead code detection...

📦 Running ts-prune...
   Found 654 issues

🔪 Running knip...
   Found 33 issues

scripts/detect-dead-code-ts.sh: line 277: /usr/bin/jq: Argument list too long
error: script "dead-code:ts" exited with code 126

Error: Dead code detection failed - threshold exceeded
cat: reports/dead-code-report.json: No such file or directory
Error: Process completed with exit code 1.
```

**Total issues**: 687 (654 from ts-prune + 33 from knip)

---

## Root Cause

At `scripts/detect-dead-code-ts.sh:259`, the script uses command substitution to pass the entire issues JSON file as a command-line argument to `jq`:

```bash
REPORT_JSON=$(jq -n \
  --arg timestamp "$TIMESTAMP" \
  # ... other args ...
  --argjson issues "$(cat "$ISSUES_JSON")" \    # ← Problem: passes entire file as argument
  '{
    timestamp: $timestamp,
    # ... build report ...
    issues: $issues,
  }')
```

With 687 issues, the JSON file is large enough to exceed the system's `ARG_MAX` limit (typically 128KB-2MB on Linux systems), causing the error:

```
/usr/bin/jq: Argument list too long
```

**Impact:**
- Script exits with code 126 (command cannot execute)
- Report JSON file is never created
- Subsequent steps fail trying to read the non-existent report
- CI job fails

---

## Solution

Replace command substitution with `--slurpfile` to read the file directly instead of passing it as an argument:

```bash
REPORT_JSON=$(jq -n \
  --arg timestamp "$TIMESTAMP" \
  --argjson totalIssues "$TOTAL_ISSUES" \
  --argjson tsPrune "$TS_PRUNE_TOTAL" \
  --argjson knip "$KNIP_TOTAL" \
  --argjson unusedExports "$UNUSED_EXPORTS" \
  --argjson unusedFiles "$UNUSED_FILES" \
  --argjson unusedDependencies "$UNUSED_DEPS" \
  --argjson other "$OTHER" \
  --slurpfile issues "$ISSUES_JSON" \              # ← Fix: read file directly
  '{
    timestamp: $timestamp,
    totalIssues: $totalIssues,
    byTool: {
      tsPrune: $tsPrune,
      knip: $knip
    },
    byType: (
      $issues[0] | group_by(.type) | map({key: .[0].type, value: length}) | from_entries
    ),
    issues: $issues[0],                             # ← Access first element (slurpfile wraps in array)
    summary: {
      unusedExports: $unusedExports,
      unusedFiles: $unusedFiles,
      unusedDependencies: $unusedDependencies,
      other: $other
    }
  }')
```

**Key changes:**
1. Replace `--argjson issues "$(cat "$ISSUES_JSON")"` with `--slurpfile issues "$ISSUES_JSON"`
2. Access `$issues[0]` in the jq expression (slurpfile wraps the file content in an array)
3. Update both references: `$issues[0] | group_by(...)` and `issues: $issues[0]`

**Why this works:**
- `--slurpfile` reads the file directly via a file descriptor, avoiding ARG_MAX limits
- No command substitution means no large string passed as argument
- Can handle JSON files of any size (limited only by available memory/disk)

---

## Alternative Solutions

If `--slurpfile` is not available in the jq version:

### Option 1: Use stdin with jq

```bash
REPORT_JSON=$(jq -n \
  --arg timestamp "$TIMESTAMP" \
  # ... other args ...
  "$(cat <<JQSCRIPT
{
  timestamp: \$timestamp,
  # ... rest of report structure ...
  issues: $(cat "$ISSUES_JSON")
}
JQSCRIPT
)")
```

### Option 2: Two-pass approach

```bash
# First pass: create report without issues
REPORT_JSON=$(jq -n \
  --arg timestamp "$TIMESTAMP" \
  # ... other args ...
  '{
    timestamp: $timestamp,
    # ... summary fields ...
  }')

# Second pass: merge with issues
REPORT_JSON=$(echo "$REPORT_JSON" | jq --slurpfile issues "$ISSUES_JSON" '. + {issues: $issues[0]}')
```

---

## Testing

To reproduce locally:
```bash
# This will likely trigger the bug if you have many dead code issues
bash scripts/detect-dead-code-ts.sh --output-dir=reports --threshold=10
```

After fix, verify:
1. Script completes successfully with high issue counts
2. `reports/dead-code-report.json` is created and valid
3. All issues are included in the report
4. CI job passes

---

## Acceptance Criteria

- [ ] Script handles 687+ issues without "Argument list too long" error
- [ ] Report JSON file is created successfully
- [ ] All detected issues are included in the report
- [ ] CI job completes successfully
- [ ] Solution tested with both low (<100) and high (>500) issue counts

---

## Additional Context

This is a common shell scripting pitfall when dealing with large data:
- `ARG_MAX` limit varies by system (2MB on modern Linux, 256KB on older systems)
- Command substitution `$(...)` passes output as argument, subject to ARG_MAX
- File operations (`--slurpfile`, `<`, etc.) bypass ARG_MAX by using file descriptors

**Related improvement**: Consider streaming/chunking the report generation if memory becomes an issue with thousands of issues.

---

## Files to Modify

- `scripts/detect-dead-code-ts.sh` (line 250-277)

--- a/scripts/detect-dead-code-ts.sh
+++ b/scripts/detect-dead-code-ts.sh
@@ -256,7 +256,7 @@ REPORT_JSON=$(jq -n \
   --argjson unusedFiles "$UNUSED_FILES" \
   --argjson unusedDependencies "$UNUSED_DEPS" \
   --argjson other "$OTHER" \
-  --argjson issues "$(cat "$ISSUES_JSON")" \
+  --slurpfile issues "$ISSUES_JSON" \
   '{
     timestamp: $timestamp,
     totalIssues: $totalIssues,
@@ -265,9 +265,9 @@ REPORT_JSON=$(jq -n \
       knip: $knip
     },
     byType: (
-      $issues | group_by(.type) | map({key: .[0].type, value: length}) | from_entries
+      $issues[0] | group_by(.type) | map({key: .[0].type, value: length}) | from_entries
     ),
-    issues: $issues,
+    issues: $issues[0],
     summary: {
       unusedExports: $unusedExports,
       unusedFiles: $unusedFiles,
@@ -323,8 +323,8 @@ else
                 echo "📍 $type_upper ($count):"
                 echo "────────────────────────────────────────────────────────────"
 
-                # Show first 20 issues
-                jq -r ".[] | select(.type == \"$type\") | \"  \(.location) - \(.name)\"" "$ISSUES_JSON" | head -20
+                # Show first 20 issues without triggering pipefail on SIGPIPE
+                jq -r --arg type "$type" 'map(select(.type == $type)) | .[:20][] | "  \(.location) - \(.name)"' "$ISSUES_JSON"
 
                 if [ "$count" -gt 20 ]; then
                     echo "  ... and $((count - 20)) more"

